sound/soc/sof/topology.c

```diff
@@ -1233,6 +1243,60 @@ static int sof_control_unload(struct snd_soc_component *scomp,
  * DAI Topology
  */
 
+static int sof_core_enable(struct snd_sof_dev *sdev, int core)
+{
+	struct sof_ipc_pm_core_config pm_core_config = {
+		.hdr = {
+			.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CORE_ENABLE,
+			.size = sizeof(pm_core_config),
+		},
+		.enable_mask = sdev->enabled_cores_mask | BIT(core),
+	};
+	int ret;
+
```

Comment:
error check if core isn't in the range of platform-supported cores? we have a cores_mask field for Intel.

Reply:
this should be done, yes, but I think this should be done inside `.core_power_up()` which is called just above and is arch-dependent, whereas this function is not. E.g. such a check should be added to `hda_dsp_enable_core()` but that should be a separate patch, since this is also relevant now, this isn't something new, added by this PR.

Reply:
ok